### PR TITLE
[DOCS] Fix get snapshot status API parameters

### LINKED ISSUE #81600: [DOCS] Fix get snapshot status API docs
As pointed out in https://github.com/elastic/elasticsearch/issues/80931, the [get snapshot status API docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-snapshot-status-api.html#get-snapshot-status-api-desc) have some issues:

* `GET _snapshot/_status` `GET _snapshot/<repository>/_status` are undocumented.
* The `<repository>` and `<snapshot>` parameters are not required.
* The `<snapshot>` parameter doesn't support `_current`.

Relates to https://github.com/elastic/elasticsearch/pull/59355.

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes

* Documents `GET _snapshot/_status` and `GET _snapshot/<repository>/_status`.
* Notes the `<repository>` and `<snasphot>` parameters are optional.
* Removes erroneous mention of the `<snapshot>` parameter supporting the `_current` value.

Closes #81600

Relates to #80931

### Preview
https://elasticsearch_81818.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-snapshot-status-api.html

### CODE REVIEW

File: docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc [hunk 1]
```diff
@@ -112,18 +121,14 @@ and, when using cloud storage, incur high processing costs.
 ==== {api-path-parms-title}
 
 `<repository>`::
-(Required, string)
-Snapshot repository name used to limit the request. Wildcard (`*`) expressions are supported.
-+
-Use `_current` after the repository name to limit the request only to the currently running snapshots. This usage is preferred to return only snapshots that are currently running and not degrade response time.
+(Optional, string)
+Snapshot repository name used to limit the request. Wildcards (`*`) are not
+supported.
```
Comment (jrodewig) [inline docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc:126]: This is actually a little bit tricky. The current docs indicate wildcards are supported. However, using something like `*` or `my_repo*` doesn't resolve. `_all` (the default) only works with `GET _snapshot/_all/_status`. However, that's equivalent to `GET _snapshot/_status`.

Let me know if we'd like to document `_all` given the caveats. Thanks!

**Comment by arteam** [comment]
@elasticmachine update branch

File: docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc]: Actually I'm wrong. This supports wildcards if the `<snapshot>` isn't specified.

**Comment by jrodewig** [comment]
@arteam @original-brownbear Do you mind taking a look at this PR when you have a moment? Thanks!

**Review by arteam — APPROVED** [review]

**Comment by arteam** [comment]
The changes look good to me. Thank James!

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/83460)|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/83461)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/83462)|
|✅|7.16|[](https://github.com/elastic/elasticsearch/pull/83463)|
|✅|7.15|[](https://github.com/elastic/elasticsearch/pull/83464)|
|✅|7.14|[](https://github.com/elastic/elasticsearch/pull/83465)|
|✅|7.13|[](https://github.com/elastic/elasticsearch/pull/83466)|
|✅|7.12|[](https://github.com/elastic/elasticsearch/pull/83467)|
|✅|7.11|[](https://github.com/elastic/elasticsearch/pull/83468)|
|✅|7.10|[](https://github.com/elastic/elasticsearch/pull/83469)|
|✅|7.9|[](https://github.com/elastic/elasticsearch/pull/83470)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**